app/src/main/java/eu/theinvaded/mastondroid/ui/activity/LoginActivity.java

```diff
@@ -70,49 +94,67 @@ public void startMainActivity() {
     }
 
     @Override
-    public String getClientSecret() {
-        return getString(R.string.clientSecret);
-    }
-
-    @Override
-    public String getClientId() {
-        return getString(R.string.clientId);
-    }
-
-    @Override
     public void setUser(MastodonAccount account) {
         preferences
                 .edit()
                 .putString(getString(R.string.CURRENT_USERNAME), account.username)
                 .apply();
     }
 
-
-
     @Override
-    public void showLoginError() {
-        Toast.makeText(context, R.string.login_failed_message, Toast.LENGTH_SHORT).show();
+    public void signIn(String node) {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("client_id", preferences.getString(getString(R.string.clientId), ""));
+        parameters.put("redirect_uri", "eu.theinvaded.mastondroid://oauth2redirect/");
+        parameters.put("response_type", "code");
+        parameters.put("scope", "read write follow");
+        String queryParameters;
+        try {
+            queryParameters = toQueryString(parameters);
+        } catch (UnsupportedEncodingException e) {
+            //TODO: No clue how to handle this error case??
+            assert(false);
+            return;
+        }
+        String url = "https://mastodon.social/oauth/authorize?" + queryParameters;
+        Intent viewIntent = new Intent("android.intent.action.VIEW", Uri.parse(url));
+        startActivity(viewIntent);
     }
 
     @Override
-    public void setNoUsernameError() {
-        binding.usernameLayout.setErrorEnabled(true);
-        binding.usernameLayout.setError(getString(R.string.NO_USERNAME_ERROR));
+    public boolean checkAppRegistered() {
+        return preferences.contains(getString(R.string.clientId))
+                && preferences.contains(getString(R.string.clientSecret));
```

Comment:
A bit more readable since the IDE already replaces the getString(R.string.something) with the value from the strings.xml.

The refactoring required seems extensive and outside this feature's scope. Maybe we'll make a separate issue out of simplifying working with procedures?